File: test_regress/t/t_flag_suggest.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2008 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(vlt => 1);

my @opts;
# Typo
push @opts, "-ccc";
# Typo of an option that starts with "--"
push @opts, "--ccc";
# Typo of an option that starts with "-no-"
push @opts, "-no-asserT";
# Typo of an option that starts with "-no"
push @opts, "-noasserT";
# Typo of an option that allows "-no"
push @opts, "-asserT";
# Typo of an option that starts with '+'
push @opts, "+definE+A=B";
# Typo that takes arg
push @opts, "-CFLAGs -ggdb";
# Typo of an undocumented option
push @opts, "-debug-aborT";
# Typo of "-Wno" for partial match
push @opts, "-Won-SPLITVAR";
# Typo after -Wno- for partial match
push @opts, "-Wno-SPLITVER";
# Typo of -language
push @opts, "-language 1364-1997";

my $cmd = "";

foreach my $var (@opts) {
    $cmd = $cmd . $ENV{VERILATOR_ROOT} . "/bin/verilator ${var}; ";
}

run(cmd => [$cmd],
    verilator_run => 1,
    logfile => "$Self->{obj_dir}/sim.log",
    fails => 1,
    expect_filename => $Self->{golden_filename},
);

ok(1);
1;

File: test_regress/t/t_flag_suggest.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt')

opts = []
# Typo
opts += ["-ccc"]
# Typo of an option that starts with "--"
opts += ["--ccc"]
# Typo of an option that starts with "-no-"
opts += ["-no-asserT"]
# Typo of an option that starts with "-no"
opts += ["-noasserT"]
# Typo of an option that allows "-no"
opts += ["-asserT"]
# Typo of an option that starts with '+'
opts += ["+definE+A=B"]
# Typo that takes arg
opts += ["-CFLAGs -ggdb"]
# Typo of an undocumented option
opts += ["-debug-aborT"]
# Typo of "-Wno" for partial match
opts += ["-Won-SPLITVAR"]
# Typo after -Wno- for partial match
opts += ["-Wno-SPLITVER"]
# Typo of -language
opts += ["-language 1364-1997"]

cmd = ""
for var in opts:
    cmd += os.environ["VERILATOR_ROOT"] + "/bin/verilator " + var + "; "

test.run(cmd=[cmd],
         verilator_run=True,
         logfile=test.obj_dir + "/sim.log",
         fails=True,
         expect_filename=test.golden_filename)

test.passes()
